lib/matplotlib/tests/test_axes.py

```diff
@@ -145,6 +145,24 @@ def test_twinx_cla():
     assert ax.yaxis.get_visible()
 
 
+@cleanup
+def test_twin_inherit_autoscale():
```

Comment:
Yeah, I could create something motivated by the example in #6860 (i.e. that the plotted points should reflect that the original (visible) axis and twinned (invisible) axis share the same autoscaling behavior.